theia-extensions/launcher/src/node/desktopfile-endpoint.ts

```diff
@@ -0,0 +1,151 @@
+/********************************************************************************
+ * Copyright (C) 2024 STMicroelectronics and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ *
+ * SPDX-License-Identifier: MIT
+ ********************************************************************************/
+
+import { BackendApplicationContribution } from '@theia/core/lib/node/backend-application';
+import { Application, Router } from '@theia/core/shared/express';
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { Request, Response } from 'express-serve-static-core';
+import { json } from 'body-parser';
+import { EnvVariablesServer } from '@theia/core/lib/common/env-variables';
+import { getStorageFilePath } from './launcher-util';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+
+interface DesktopFileInformation {
+    appImage: string;
+    declined: string[];
+}
+
+@injectable()
+export class TheiaDesktopFileServiceEndpoint implements BackendApplicationContribution {
```

Comment:
Why does this have to be a separate endpoint? Can't we just make it a back-end service?
Also: can we scope this to linux?

Reply:
I wanted to be consistent with the existing code in that package that may create a launcher script that is available on the path. If we refactor this, then we should probably update both.